core/deployment/src/main/java/io/quarkiverse/langchain4j/deployment/MethodUtil.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkiverse.langchain4j.deployment;
+
+import org.jboss.jandex.MethodInfo;
+
+public final class MethodUtil {
```

Comment:
I am wondering if there is already a util like this in Quarkus

Reply:
Maybe `Methods` has something like it. Looking.

Reply:
Didn't find any.

Reply:
I'll get back to you in a few minutes

Reply:
Ok, meanwhile I added a couple of tests:
- one the verifies the problem
- one the verifies the correctness of the util

Reply:
> I'll get back to you in a few minutes

Does `io.quarkus.gizmo.SignatureBuilder` do the trick?

Reply:
Let me check

Reply:
No, I don't think we can use that.

They don't seem to provide the name and that makes it of limited use to our use case: https://github.com/quarkusio/gizmo/blob/main/src/main/java/io/quarkus/gizmo/MethodSignatureBuilderImpl.java